crates/bdk/src/descriptor/dsl.rs

```diff
@@ -824,16 +824,16 @@ mod test {
         for i in 0..expected.len() {
             let index = i as u32;
             let child_desc = if !desc.has_wildcard() {
-                desc.at_derivation_index(0)
+                desc.at_derivation_index(0).unwrap()
             } else {
-                desc.at_derivation_index(index)
+                desc.at_derivation_index(index).unwrap()
             };
```

Comment:
```rust
let child_desc = desc
    .at_derivation_index(i as u32)
    .expect("must not be hardened");
```

I think this can be simplified to this. For non-wildcard or single keys, `at_derivation_index` basically just wraps the original key in `DefiniteDescriptorKey` (and `index` is not used anyway).